template/src/main/resources/application.yml

```diff
@@ -9,27 +9,21 @@
     {%- endif -%}
 {%- endfor -%}
 
-{%- for serverName, server in asyncapi.servers() %}{% if server.protocol() == 'amqp' %}
+  {%- for serverName, server in asyncapi.servers() %}{% if server.protocol() == 'amqp' %}
 amqp:
   broker: {% for line in server.description() | splitByLines %}
     # {{line | safe}}{% endfor %}
-    host: {{server.url() | replace(':{port}', '') }}
+    host: {% if server.variable('port') %}{{server.url() | replace('{port}', server.variable('port').defaultValue())}}{% else %}{{server.url()}}{% endif %}
     port: {% if server.variable('port') %}{{server.variable('port').defaultValue()}}{% endif %}
-    username:
+    username: {% if server.variable('username') %}{{server.variable('username').defaultValue()}}{% endif %}
     password:
-  exchange:
-  {% for channelName, channel in asyncapi.channels() %}
-    {% if channel.hasSubscribe() and channel.subscribe().binding('amqp') %}
-    {{channelName}}: {{channel.subscribe().binding('amqp').exchange.name}}
-    {% endif %}
-  {% endfor %}
-  queue:
   {% for channelName, channel in asyncapi.channels() %}
-    {% if channel.hasPublish() and channel.publish().binding('amqp') %}
-    {{channelName}}: {{channel.publish().binding('amqp').queue.name}}
-    {% endif %}
+  {{channelName}}:
+    exchange: {{channel.publish().binding('amqp').exchange}}
+    queue: {{channel.subscribe().binding('amqp').queue}}
+    routingKey: {{channel.subscribe().binding('amqp').routingKey}}
```

Comment:
I'm not sure that this is a good idea to generate exchange and queue for all channels without taking into account if API really has publish\subscribe for this channel

Reply:
@Tenischev So would the best way to go about it be to put in a check to see if the publish/ subscribe has been defined for that channel and then define exchange/queues etc?

Reply:
I think yes, of course this is not so critical for the configuration file, but it will keep it clean